embed/serve.go

```diff
@@ -122,13 +122,19 @@ func (sctx *serveCtx) serve(s *etcdserver.EtcdServer, tlscfg *tls.Config, handle
 // grpcHandlerFunc returns an http.Handler that delegates to grpcServer on incoming gRPC
 // connections or otherHandler otherwise. Copied from cockroachdb.
 func grpcHandlerFunc(grpcServer *grpc.Server, otherHandler http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
+	if otherHandler != nil {
```

Comment:
probably just have:
```go
if otherHandler == nil {
    return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
        grpcServer.ServeHTTP(w, r)
    })
}
return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
...
```
to avoid the extra indent level and keep the diff smaller